include/common.php

```diff
@@ -39,7 +45,24 @@ function config( $key ) {
 			'hsa_src', 'hsa_ad', 'hsa_acc', 'hsa_net', 'hsa_kw', 'hsa_tgt',
 			'hsa_ver', '_branch_match_id',
 		],
-	][ $key ];
+
+		// Add items to this array to add a unique cache variant.
+		'variants' => [],
+	];
+
+	// Run a custom configuration file.
+	if ( defined( 'WP_CACHE_CONFIG' ) ) {
+		$_config = ( function( $config ) {
+			$_config = (array) include( WP_CACHE_CONFIG );
+			return $_config;
+		} ) ( $config );
+
+		foreach ( $_config as $_key => $_value ) {
```

Comment:
Hmm, I thought foreach was faster since you didn't have any of the additional key check logic (string vs numeric). I was wrong, at least for PHP 8.1:

![image](https://user-images.githubusercontent.com/108344/145978390-47f84078-cfca-4270-8290-8f059e053953.png)

Reply:
splitting hairs, but true.

`yield` (https://www.php.net/manual/en/language.generators.overview.php) would be even better, especially if the array will increase. But for now, can't image any improvement with yield.

Reply:
Nice benchmark test by the way!